File: Interviewbit/LinkedList/Merge Two Sorted Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
ListNode* Solution::mergeTwoLists(ListNode* A, ListNode* B) {
    
    ListNode* head;
    
    if(B->val <A->val)
    {
        head=B;
        B=B->next;
    }
    else
    {
        head=A;
        A=A->next;
    }
    ListNode* temp=head;
    while(A&&B)
    {
        if(A->val<B->val)
        {
            temp->next=A;
            A=A->next;
           
        }
        else 
        {
            temp->next=B;
            B=B->next;
           
        }
        temp=temp->next;     
    }
    
         if(A)
            temp->next=A;
         
         if(B)
            temp->next=B;
                    
    return head;
}

File: LeetCode problems/LinkedList/Merge Two Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        
        ListNode dummy= new ListNode();
        ListNode head=dummy;
        while(l1 != null && l2 != null)
        {
            if(l1.val<l2.val)
            {
                dummy.next=l1;
                l1=l1.next;
                
            }
            else
            {
                dummy.next=l2;
                l2=l2.next;
            }
            dummy=dummy.next;
        }
        if(l1!=null)
        {
            dummy.next=l1;
        }
        else
            dummy.next=l2;
        
        return head.next;
    }
    
}